impeller/entity/render_target_cache.cc

```diff
@@ -44,7 +48,13 @@ RenderTarget RenderTargetCache::CreateOffscreen(
     const auto other_config = render_target_data.config;
     if (!render_target_data.used_this_frame && other_config == config) {
       render_target_data.used_this_frame = true;
-      return render_target_data.render_target;
+      auto color0 = render_target_data.render_target.GetColorAttachments()
+                        .find(0u)
+                        ->second;
+      auto depth = render_target_data.render_target.GetDepthAttachment();
+      return RenderTargetAllocator::CreateOffscreen(
+          context, size, mip_count, label, color_attachment_config,
+          stencil_attachment_config, color0.texture, depth->texture);
```

Comment:
This is getting flagged as unchecked optional usage, since depth may not be defined - but that should be OK if you default depth.texture to nullptr